cpp/recognition/planar/HopcroftTarjan.cpp

```diff
@@ -0,0 +1,303 @@
+                }
+            }
+        }
+    }
+
+
+    void HopcroftTarjan::reorder(NetworKit::Graph &g, 
+                std::vector<NetworKit::node> & parent, 
+                std::vector<NetworKit::node> & dfsnum,
+                std::vector<NetworKit::Edge> & order
+                )
+    {
+        NetworKit::node v = 0;
+        std::vector<bool> reached(N);
+        int dfs_count = 0;
+        std::vector<NetworKit::Edge> Add;
+        std::vector<NetworKit::node> lowpt(N), lowpt2(N);
+        dfs_in_reorder(g, v, Add, dfs_count, reached, dfsnum, lowpt, lowpt2, parent);
+        std::vector<NetworKit::Edge> buckets[N*2+5];
+        for(auto u: Add){
+            int cost = ((dfsnum[u.v] < dfsnum[u.u]) ? 2 * dfsnum[u.v] :
+            ((lowpt2[u.v] >= dfsnum[u.u]) ? 2 * lowpt[u.v] : 2 * lowpt[u.v] + 1));
+            buckets[cost].push_back(u);
+        }
+        g = NetworKit::Graph(N, true, true);
+        order = std::vector<NetworKit::Edge>(Add.size());
+        int id = 0;
+        for(int i = 0; i <= 2*N; i++){
+            for(auto u: buckets[i]){
+                g.addEdge(u.u, u.v, id);
+                order[id] = NetworKit::Edge(u.u, u.v);
+                id++;
+            }
+        }
+    }
+
+    bool HopcroftTarjan::strong_planar(NetworKit::Graph &g, 
+                       NetworKit::Edge e, 
+                       std::list<int> & Att,
+                       std::vector<int> & alpha,
+                       std::vector<NetworKit::node> & dfsnum,
+                       std::vector<NetworKit::node> & parent
+                    )
+    {
+        NetworKit::node  y = e.v, t = *g.neighborRange(y).begin(), x = e.u, wk = y;
+        while(dfsnum[t] > dfsnum[wk]){
+            wk = t;
+            t = *g.neighborRange(wk).begin();
+        }
+        NetworKit::node w0 = t;
+        NetworKit::node w = wk;
+        std::stack<Block*> S;
+        while(w != x) {
+            int count = 0;
+            for(auto u : g.weightNeighborRange(w)) {
+                count++;
+                if(count != 1) {
+                    std::list<int> A;
+                    if(dfsnum[w] < dfsnum[u.first]) {
+                        if(!strong_planar(g, NetworKit::Edge(w, u.first), A, alpha, dfsnum, parent)){
+                            while(!S.empty()){
+                                delete S.top();
+                                S.pop();
+                            } 
+                            return false;
+                        }
+                    } else {
+                        A.push_back(dfsnum[u.first]);
+                    }
+
+                    Block * b = new Block(NetworKit::WeightedEdge(w, u.first, u.second), A);
+                    while(true){
+                        if(b->left_interlace(S)){
+                            (S.top())->flip();
+                        }
+                        if(b->left_interlace(S)){
+                            delete b;
```

Comment:
In general, please avoid using pointers and new/delete unless it's really necessary